### ISSUE TITLE
Sort output by file line order rather than rule number

### ISSUE THREAD

**Issue by JayBeeDe** [markdownlint/markdownlint#416]
## Describe the Enhancement:
Add a CLI parameter such as "-S" flag in order to require the output to be sorted by file line number rather than line reference.

### Impacted Rules:
All rules

### Current implementation example
```none
mdl README.md -r "~MD033,~MD013"
README.md:6: MD004 Unordered list style
README.md:7: MD004 Unordered list style
README.md:10: MD004 Unordered list style
README.md:11: MD004 Unordered list style
README.md:12: MD004 Unordered list style
README.md:13: MD004 Unordered list style
README.md:161: MD004 Unordered list style
README.md:162: MD004 Unordered list style
README.md:175: MD004 Unordered list style
README.md:176: MD004 Unordered list style
README.md:177: MD004 Unordered list style
README.md:179: MD004 Unordered list style
README.md:62: MD029 Ordered list item prefix
README.md:71: MD029 Ordered list item prefix
README.md:74: MD029 Ordered list item prefix
README.md:77: MD029 Ordered list item prefix
README.md:80: MD029 Ordered list item prefix
README.md:71: MD032 Lists should be surrounded by blank lines
README.md:74: MD032 Lists should be surrounded by blank lines
README.md:77: MD032 Lists should be surrounded by blank lines
README.md:80: MD032 Lists should be surrounded by blank lines

A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md
```
### Target implementation example
```none
mdl README.md -r "~MD033,~MD013" -S
README.md:6: MD004 Unordered list style
README.md:7: MD004 Unordered list style
README.md:10: MD004 Unordered list style
README.md:11: MD004 Unordered list style
README.md:12: MD004 Unordered list style
README.md:13: MD004 Unordered list style
README.md:62: MD029 Ordered list item prefix
README.md:71: MD029 Ordered list item prefix
README.md:71: MD032 Lists should be surrounded by blank lines
README.md:74: MD029 Ordered list item prefix
README.md:74: MD032 Lists should be surrounded by blank lines
README.md:77: MD029 Ordered list item prefix
README.md:77: MD032 Lists should be surrounded by blank lines
README.md:80: MD029 Ordered list item prefix
README.md:80: MD032 Lists should be surrounded by blank lines
README.md:161: MD004 Unordered list style
README.md:162: MD004 Unordered list style
README.md:175: MD004 Unordered list style
README.md:176: MD004 Unordered list style
README.md:177: MD004 Unordered list style
README.md:179: MD004 Unordered list style

A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md
```

## Current Alternative
Using the bash sort function:
```bash
mdl README.md -r "~MD033,~MD013" | sort -V
```

**Comment by jaymzh** [comment]
Seems reasonable. I'd accept a PR for that.
